src/library/bpmdelegate.cpp

```diff
@@ -75,6 +77,21 @@ void BPMDelegate::paint(QPainter* painter,const QStyleOptionViewItem &option,
     QStyleOptionViewItemV4 opt = option;
     initStyleOption(&opt, index);
 
+    // Set the palette appropriately based on whether the row is selected or
```

Comment:
These lines are duplicated in TableItemDelegate::paint(). Can we move them into a protected utility function in the base class?